cmd/vulcan-nuclei/main.go

```diff
@@ -0,0 +1,362 @@
+		"-H", userAgent,
+	}
+
+	// Include only selected severities.
+	selectedSeverities := strings.Join(opt.Severities, ",")
+	if len(opt.Severities) > 0 {
+		logger.Infof("selected severities: %s", selectedSeverities)
+		severities := []string{"-severity", selectedSeverities}
+		nucleiArgs = append(nucleiArgs, severities...)
+	}
+
+	logger.Debugf("nuclei command args: [%s]", nucleiArgs)
+	var nucleiVulns []ResultEvent
+	for t := range selectedTemplates {
+		nucleiArgsWithTemplate := append(nucleiArgs, "-t", t)
+		output, err := runNucleiCmd(nucleiArgsWithTemplate)
+		if err != nil {
+			logger.Errorf("nuclei execution failed with template [%s]: %s", t, err)
+			continue
+		}
+		if len(output) == 0 {
+			logger.Infof("no vulnerabilities found with template [%s]", t)
+			continue
+		}
+		scanner := bufio.NewScanner(bytes.NewReader(output))
+		for scanner.Scan() {
+			var v ResultEvent
+			err := json.Unmarshal(scanner.Bytes(), &v)
+			if err != nil {
+				logger.Errorf("unable to unmarshal vulnerability [%s]: %s", scanner.Text(), err)
+				continue
+			}
+			nucleiVulns = append(nucleiVulns, v)
+		}
+	}
+	// No vulnerabilities found. Return.
+	if len(nucleiVulns) < 1 {
+		logger.Info("no vulnerabilities found")
+		return nil
+	}
+	logger.Infof("nuclei found [%d] vulnerabilities", len(nucleiVulns))
+
+	rv := make(map[string]*report.Vulnerability)
+	for _, v := range nucleiVulns {
+		// Create resources table row.
+		// Avoid store redundant information in the resources table.
+		matched := v.Matched
+		if matched == target {
+			matched = ""
+		}
+		findingRow := map[string]string{
+			"Template":    v.TemplateID,
+			"MatcherName": v.MatcherName,
+			"Matched":     matched,
+		}
+		extractedResults := false
+		if len(v.ExtractedResults) > 0 {
+			extractedResults = true
+		}
+		resultRow := map[string]string{
+			"References": strings.Join(v.ExtractedResults, "<br>"),
+		}
+		if vf, ok := rv[v.TemplateID]; ok {
+			vf.Resources[0].Rows = append(vf.Resources[0].Rows, findingRow)
+			if extractedResults {
+				vf.Resources[1].Rows = append(vf.Resources[1].Rows, resultRow)
+			}
+			continue
+		}
+		recommendations := []string{v.Info.Remediation}
+		if v.Info.Remediation == "" {
+			recommendations = []string{
+				"The check does not provide specific recommendations for this issue.",
+				"Take a look to reference links (if any) for further details about the finding.",
+			}
+		}
+		var vuln = report.Vulnerability{
+			AffectedResource: target,
+			CWEID:            getCWEID(v.Info.Classification.CWEID),
+			Summary:          v.Info.Name,
+			Description:      v.Info.Description,
+			Details:          generateDetails(target, v.Template),
+			Score:            getScore(v.Info.Severity),
+			References:       v.Info.Reference,
+			Recommendations:  recommendations,
+			Labels:           []string{"nuclei", v.Type},
+		}
+		if strings.EqualFold(v.Info.Severity, "info") {
+			vuln.Labels = append(vuln.Labels, "informational")
+		} else {
```

Comment:
I think we can remove this as agreed in the discussion we had the other day.

Reply:
Applied suggestions here [9ab8f84](https://github.com/adevinta/vulcan-checks/pull/86/commits/9ab8f84a5b476aa4efa174ed27feaf90832aca4d).